doc: doxygen's automatic file matching files

Currently we get a large number for warning like this when building doxygen:
```
/tmp/RIOT/cpu/atmega1281/include/cpu_conf.h:12: warning: the name `/tmp/RIOT/cpu/atmega1281/include/cpu_conf.h' supplied as the second argument in the \file statement matches the following input files:
   /tmp/RIOT/cpu/atmega1281/include/cpu_conf.h
Please use a more specific name by including a (larger) part of the path!
```
This looks IMO strange since apparently only one matching input file was found, so no problem should exist. Of course, the simple solution could be to specify explicitly things like
```
@file cpu/atmega1281/include/cpu_conf.h
```
for each CPU and board, but I thought doxygen should be able to resolve this itself.

Indeed:
```/home/oleg/git/RIOT/cpu/atmega1281/include/cpu_conf.h:12: warning: the name `cpu/atmega1281/include/cpu_conf.h' supplied as the second argument in the \file statement is not an input file
```
if I try to use `cpu/atmega1281/include/cpu_conf.h`.

--- a/doc/doxygen/Makefile
+++ b/doc/doxygen/Makefile
@@ -22,4 +22,4 @@ latex:
 	( cat riot.doxyfile ; echo "GENERATE_LATEX= yes" ) | doxygen -
 
 clean:
-	-@rm -rf latex man html doxygen_objdb_*.tmp
+	-@rm -rf latex man html doxygen_objdb_*.tmp doxygen_entrydb_*.tmp

--- a/doc/doxygen/Makefile
+++ b/doc/doxygen/Makefile
@@ -1,7 +1,7 @@
 RIOTBASE=$(shell git rev-parse --show-toplevel)
 # Generate list of quoted absolute include paths. Evaluated in riot.doxyfile.
 export STRIP_FROM_INC_PATH_LIST=$(shell \
-    git ls-tree -dr --full-tree --name-only HEAD |\
+    git ls-tree -dr --full-tree --name-only HEAD core drivers sys |\
     grep '/include$$' |\
     sed 's/.*/\"$(subst /,\/,${RIOTBASE})\/\0\"/')
 

--- a/doc/doxygen/riot.doxyfile
+++ b/doc/doxygen/riot.doxyfile
@@ -841,6 +841,9 @@ EXCLUDE_PATTERNS       = */board/*/tools/* \
                          */sys/random/tinymt32/* \
                          */cpu/stm32f2/include/stm32f2*.h \
                          */cpu/stm32f4/include/stm32f4*.h \
+                         */cpu/stellaris_common/include/stellaris_periph \
+                         */cpu/native/osx-libc-extra \
+                         */toolchain/* \
 
 
 # The EXCLUDE_SYMBOLS tag can be used to specify one or more symbol names

